core/base/inc/TSystem.h

```diff
@@ -414,24 +410,25 @@ class TSystem : public TNamed {
    virtual void           *OpenDirectory(const char *name);
    virtual void            FreeDirectory(void *dirp);
    virtual const char     *GetDirEntry(void *dirp);
-   virtual void           *GetDirPtr() const { return 0; }
+   virtual void           *GetDirPtr() const { return nullptr; }
    virtual Bool_t          ChangeDirectory(const char *path);
    virtual const char     *WorkingDirectory();
    virtual std::string     GetWorkingDirectory() const;
-   virtual const char     *HomeDirectory(const char *userName = 0);
-   virtual std::string     GetHomeDirectory(const char *userName = 0) const;
+   virtual const char     *HomeDirectory(const char *userName = nullptr);
+   virtual std::string     GetHomeDirectory(const char *userName = nullptr) const;
    virtual int             mkdir(const char *name, Bool_t recursive = kFALSE);
    Bool_t                  cd(const char *path) { return ChangeDirectory(path); }
    const char             *pwd() { return WorkingDirectory(); }
    virtual const char     *TempDirectory() const;
-   virtual FILE           *TempFileName(TString &base, const char *dir = 0);
+   virtual FILE           *TempFileName(TString &base, const char *dir = nullptr);
 
    //---- Paths & Files
    virtual const char     *BaseName(const char *pathname);
    virtual const char     *DirName(const char *pathname);
+   virtual TString         GetDirName(const char *pathname);
```

Comment:
Consider:
```suggestion
   virtual TString         GetDirName(std::string_view pathname);
```

Reply:
Should we change type of input arguments from `const char *` to `std::string_view`? 
Should we do it for all new methods?

Reply:
In principle yes.  In practice, to make it beneficial, you also often have to change all the inner functions (that takes those strings).   So in practice, I think we should for new functions and for hotspots.

Reply:
Replacing `const char *` by `std::string_view` does not increase performance.
It only makes sense if caller really operates with `string_view` instances to avoid possible intermediate copies.

Reply:
Indeed. std::string_view will only improve the performance if the callees is update to leverage it (by avoiding copying and/or updates (like c_str() does)).